src/argilla/monitoring/asgi.py

```diff
@@ -67,16 +67,52 @@ def token_classification_mapper(inputs, outputs):
 
 
 def text_classification_mapper(inputs, outputs):
-    return TextClassificationRecord(
-        inputs=inputs,
-        prediction=[
-            (label, score)
-            for label, score in zip(
-                outputs.get("labels", []), outputs.get("scores", [])
+    if isinstance(inputs, str):
+        return TextClassificationRecord(
+            text=inputs,
+            prediction=[
+                (label, score)
+                for label, score in zip(
+                    outputs.get("labels", []), outputs.get("scores", [])
+                )
+            ],
+            event_timestamp=datetime.datetime.now(),
+        )
+    elif isinstance(inputs, dict):
+        text = inputs.get("text")
+        if text:
```

Comment:
Here, I would include an extra condition: `len(inputs) == 1`
```suggestion
        if text and len(inputs) == 1:
```

This avoids conflicts when we have the text and other fields in the inputs